File: wagerproof-ios-native/Wagerproof/Features/Scoreboard/Components/LiveScoreCardShimmer.swift
import SwiftUI
import WagerproofDesign

/// Shimmer placeholder for `LiveScoreCard`. Reproduces the compact tile's exact
/// bounding box (50%-width grid cell: 8pt surface, hairline border, the
/// team-abbr/score row plus a centered prediction line) so the loaded-state
/// crossfade doesn't shift the grid.
///
/// Uses the unified `.shimmering()` sweep — applied to the inner placeholder
/// group, with the solid `appSurface` chrome added afterward.
struct LiveScoreCardShimmer: View {
    var body: some View {
        VStack(alignment: .leading, spacing: 4) {
            HStack(spacing: 8) {
                HStack(spacing: 4) {
                    SkeletonBlock(width: 26, height: 12)   // away abbr
                    SkeletonBlock(width: 18, height: 14)   // away score
                    SkeletonBlock(width: 18, height: 14)   // home score
                    SkeletonBlock(width: 26, height: 12)   // home abbr
                }
                Spacer()
                SkeletonBlock(width: 34, height: 10)       // period / clock
            }
            HStack {
                Spacer()
                SkeletonCapsule(width: 64, height: 10)     // prediction line
                Spacer()
            }
            .padding(.top, 4)
        }
        .padding(.vertical, 6)
        .padding(.horizontal, 10)
        .frame(maxWidth: .infinity, alignment: .leading)
        .shimmering()
        .background(Color.appSurface)
        .clipShape(RoundedRectangle(cornerRadius: 8))
        .overlay(
            RoundedRectangle(cornerRadius: 8)
                .stroke(Color.appBorder, lineWidth: 1)
        )
    }
}

#Preview {
    LazyVGrid(columns: [GridItem(.flexible(), spacing: 8), GridItem(.flexible(), spacing: 8)], spacing: 8) {
        ForEach(0..<6, id: \.self) { _ in
            LiveScoreCardShimmer()
        }
    }
    .padding()
    .background(Color.appSurface)
}

File: wagerproof-android-native/app/src/main/java/com/wagerproof/app/features/scoreboard/components/LiveScoreCardShimmer.kt
package com.wagerproof.app.features.scoreboard.components

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.unit.dp
import com.wagerproof.core.design.components.SkeletonBlock
import com.wagerproof.core.design.components.SkeletonCapsule
import com.wagerproof.core.design.components.shimmering
import com.wagerproof.core.design.tokens.AppColors

/**
 * Shimmer placeholder matching [LiveScoreCard]'s compact-tile footprint so the
 * loaded-state crossfade doesn't shift the grid. iOS `LiveScoreCardShimmer`.
 */
@Composable
fun LiveScoreCardShimmer(modifier: Modifier = Modifier) {
    val shape = RoundedCornerShape(8.dp)
    Column(
        modifier
            .fillMaxWidth()
            .clip(shape)
            .background(AppColors.appSurface)
            .border(1.dp, AppColors.appBorder, shape),
    ) {
        Column(
            Modifier.shimmering().padding(horizontal = 10.dp, vertical = 6.dp),
            verticalArrangement = Arrangement.spacedBy(4.dp),
        ) {
            Row(verticalAlignment = Alignment.CenterVertically, horizontalArrangement = Arrangement.spacedBy(8.dp)) {
                Row(horizontalArrangement = Arrangement.spacedBy(4.dp)) {
                    SkeletonBlock(width = 26.dp, height = 12.dp)
                    SkeletonBlock(width = 18.dp, height = 14.dp)
                    SkeletonBlock(width = 18.dp, height = 14.dp)
                    SkeletonBlock(width = 26.dp, height = 12.dp)
                }
                Box(Modifier.weight(1f))
                SkeletonBlock(width = 34.dp, height = 10.dp)
            }
            Row(Modifier.fillMaxWidth().padding(top = 4.dp), horizontalArrangement = Arrangement.Center) {
                Box(Modifier.weight(1f))
                SkeletonCapsule(width = 64.dp, height = 10.dp)
                Box(Modifier.weight(1f))
            }
        }
    }
}
